hs-bindgen/src-internal/HsBindgen.hs

```diff
@@ -171,22 +171,30 @@ writeTests testDir = do
   Helpers
 -------------------------------------------------------------------------------}
 
-write :: String -> Maybe FilePath -> String -> ArtefactM ()
-write _    Nothing     str = liftIO $ putStrLn str
-write what (Just path) str = do
+write :: FileOverwritePolicy -> String -> Maybe FilePath -> String -> ArtefactM ()
+write _                    _    Nothing     str = liftIO $ putStrLn str
+write fop what (Just path) str = do
       tracer <- artefactTracer <$> ask
       traceWith tracer $ RunArtefactWriteFile what path
       liftIO $ do
-        createDirectoryIfMissing True $ takeDirectory path
-        writeFile path str
+        -- Check if file exists and policy forbids overwriting
+        fileExists <- doesFileExist path
+        case fop of
+          ProtectExistingFiles
+            | fileExists ->
+            throwIO (FileAlreadyExistsException path)
```

Comment:
I think this should be an Error trace instead (but others may disagree). ArtefactM has access to the tracer, so this should not be a problem.